Sources/Apodini/Semantic Model Builder/Model/Endpoint.swift

```diff
@@ -137,19 +127,35 @@ class EndpointsTreeNode {
     private var childContainsPathParameter = false
     
     lazy var absolutePath: [_PathComponent] = {
-        var absolutePath: [_PathComponent] = []
-        collectAbsolutePath(&absolutePath)
-        return absolutePath
+        guard let parent = parent else {
+            return []
+        }
+        
+        return parent.absolutePath + [path]
     }()
     
     lazy var relationships: [EndpointRelationship] = {
-        var relationships: [EndpointRelationship] = []
-        
-        for (name, child) in nodeChildren {
-            child.collectRelationships(name: name, &relationships)
+        func collectRelationships(_ node: EndpointsTreeNode, name: String) -> [EndpointRelationship] {
+            guard self.endpoints.isEmpty else {
+                return [EndpointRelationship(name: name, destinationPath: absolutePath)]
+            }
+            
+            var relationships: [EndpointRelationship] = []
+            for (childName, child) in node.nodeChildren {
+                // as Parameter is currently inserted into the path (which will change)
+                // checking against RequestInjectable is a lazy check to determine if this is a path parameter
+                // or just a regular path component. To be adapted.
+                let name = name + (child.path is RequestInjectable ? "" : "_" + childName)
+                relationships += collectRelationships(child, name: name)
+            }
+            
+            return relationships
         }
         
-        return relationships
+        return nodeChildren.reduce([]) { result, item in
```

Comment:
Swiftlint warning here, wants to replace it with `reduce(:into)` which makes sense(?)

Reply:
I somehow prefer the more strict immutable version, but I get it: it can be more efficient 😄